drivers/clock_control/clock_control_bl60x.c

```diff
@@ -0,0 +1,919 @@
+ * When using PLL root clock, we can use either setting, when using the 32Mhz Oscillator with a
+ * uninitialized PLL, only FCLK will be available.
+ */
+static void clock_control_bl60x_uart_set_clock(bool enable, uint32_t clock, uint32_t divider)
+{
+	uint32_t tmp;
+
+	if (divider > 0x7) {
+		divider = 0x7;
+	}
+	if (clock > 1) {
+		clock = 1;
+	}
+	/* disable uart clock */
+	clock_control_bl60x_uart_set_clock_enable(false);
+
+	tmp = sys_read32(GLB_BASE + GLB_CLK_CFG2_OFFSET);
+	tmp = (tmp & GLB_UART_CLK_DIV_UMSK) | (divider << GLB_UART_CLK_DIV_POS);
+	sys_write32(tmp, GLB_BASE + GLB_CLK_CFG2_OFFSET);
+
+	tmp = sys_read32(HBN_BASE + HBN_GLB_OFFSET);
+	tmp = (tmp & HBN_UART_CLK_SEL_UMSK) | (clock << HBN_UART_CLK_SEL_POS);
+	sys_write32(tmp, HBN_BASE + HBN_GLB_OFFSET);
+
+	clock_control_bl60x_uart_set_clock_enable(enable);
+}
+
+/* Simple function to enable all peripherals for now */
+static void clock_control_bl60x_peripheral_clock_init(void)
+{
+	uint32_t regval = sys_read32(GLB_BASE + GLB_CGEN_CFG1_OFFSET);
+
+	/* enable ADC clock routing */
+	regval |= (1 << 2);
+	/* enable UART0 clock routing */
+	regval |= (1 << 16);
+	/* enable I2C0 clock routing */
+	regval |= (1 << 19);
+
+	sys_write32(regval, GLB_BASE + GLB_CGEN_CFG1_OFFSET);
+
+	clock_control_bl60x_uart_set_clock(1, 0, 0);
+}
+
+static int clock_control_bl60x_on(const struct device *dev, clock_control_subsys_t sys)
+{
+	struct clock_control_bl60x_data *data = dev->data;
+	int ret = -EINVAL;
+	uint32_t key;
+	enum bl60x_clkid oldroot;
+
+	key = irq_lock();
+
+	if ((enum bl60x_clkid)sys == bl60x_clkid_clk_crystal) {
+		if (data->crystal_enabled) {
+			ret = 0;
+		} else {
+			data->crystal_enabled = true;
+			ret = clock_control_bl60x_update_root(dev);
+			if (ret < 0) {
+				data->crystal_enabled = false;
+			}
+		}
+	} else if ((enum bl60x_clkid)sys == bl60x_clkid_clk_pll) {
+		if (data->pll_enabled) {
+			ret = 0;
+		} else {
+			data->pll_enabled = true;
+			ret = clock_control_bl60x_update_root(dev);
+			if (ret < 0) {
+				data->pll_enabled = false;
+			}
+		}
+	} else if ((int)sys == BFLB_FORCE_ROOT_RC32M) {
+		if (data->root.source == bl60x_clkid_clk_rc32m) {
+			ret = 0;
+		} else {
+			/* Cannot fail to set root to rc32m */
+			data->root.source = bl60x_clkid_clk_rc32m;
+			ret = clock_control_bl60x_update_root(dev);
+		}
+	} else if ((int)sys == BFLB_FORCE_ROOT_CRYSTAL) {
+		if (data->root.source == bl60x_clkid_clk_crystal) {
+			ret = 0;
+		} else {
+			oldroot = data->root.source;
+			data->root.source = bl60x_clkid_clk_crystal;
+			ret = clock_control_bl60x_update_root(dev);
+			if (ret < 0) {
+				data->root.source = oldroot;
+			}
+		}
+	} else if ((int)sys == BFLB_FORCE_ROOT_PLL) {
+		if (data->root.source == bl60x_clkid_clk_pll) {
+			ret = 0;
+		} else {
+			oldroot = data->root.source;
+			data->root.source = bl60x_clkid_clk_pll;
+			ret = clock_control_bl60x_update_root(dev);
+			if (ret < 0) {
+				data->root.source = oldroot;
+			}
+		}
+	} else {
+	}
```

Comment:
By removing the comment, it nows becomes a question of "why do we need an empty `else` case here?"

This function is just trying to handle different value of `sys`, and if we can't match any, it is invalid. Why don't we just do the following?

```suggestion
	} else {
		ret = -EINVAL;
	}
```

_if `sys` is one of these values, do these, else, the input is valid_

Or, remove the else case altogether if sonarqube doesn't actually complaints?

Reply:
![image](https://github.com/user-attachments/assets/7cc92bfc-1814-47c0-8168-ba370c9bcd39)

Reply:
> I guess I should've screenshotted the sonarqube page so I dont get doubted on this? 

It's just weird that SQ only triggered by some but not all of the if..else if statements that do not end with an else.